docs/_docs/sql-reference/explain-help-with-performance-problems.adoc

```diff
@@ -0,0 +1,495 @@
+Optimal plan need to be like:
+[source,sql]
+----
+Exchange
+  IndexScan
+      table: PUBLIC.T
+      index: T_COL1_COL2_IDX
+----
+We can also use **HINTS** to achieve it:
+[source,sql]
+----
+SELECT /*+ FORCE_INDEX(t_col1_col2_idx) */ id FROM t WHERE col1 = '1' AND col2 = '2';
+----
+== Unrelated sort ordering
+By default, sorted indexes store their entries in ascending order. You can adjust the ordering of a sorted index by including the options ASC or DESC.
+Lets suppose schema and related query looks like:
+[source,sql]
+----
+CREATE TABLE t1 (id INT PRIMARY KEY, col1 VARCHAR);
+CREATE TABLE t2 (id INT PRIMARY KEY, col1 VARCHAR);
+CREATE INDEX t1_col1_desc_idx ON t1(col1 DESC);
+CREATE INDEX t1_col1_idx ON t1(col1);
+CREATE INDEX t2_col1_idx ON t2(col1);
+
+SELECT t1.id as t1id, t2.id as t2id FROM t1 JOIN t2 USING (col1);
+----
+Expected that **t1_col1_idx** and **t2_col1_idx** will be involved into execution plan, but for some reasons possible plan shows:
+[source,sql]
+----
+Project
+  MergeJoin
+    Exchange
+      Sort
+          collation: [COL1 ASC]
+        IndexScan
+            index: T1_COL1_DESC_IDX
+    Exchange
+      IndexScan
+          index: T2_COL1_IDX
+----
+We can found here additional **Sort** operator over **IndexScan** it passed here for changing not satisfying ordering from near source (index), it needs to be ASC ordered.
+Extra **Sort** operator brings additional performance costs, we can avoid it also
+with help from **HINTS** side:
+[source,sql]
+----
+SELECT /*+ FORCE_INDEX(t1_col1_idx, t2_col1_idx) */ t1.id as t1id, t2.id as t2id FROM t1 JOIN t2 USING (col1);
+----
+And possible plan becomes:
+[source,sql]
+----
+Project
+  MergeJoin
+    Exchange
+      IndexScan
+          index: T1_COL1_IDX
+    Exchange
+      IndexScan
+          index: T2_COL1_IDX
+----
+== Performance impact of correlated subqueries
+SQL-99 allows for nested subqueries at nearly all places within a query.
+Ignite 3 supports nested subqueries, it can be correlated or not. Performance of certain complex correlated subqueries may be insufficient. Let`s consider a correlated query:
+[source,sql]
+----
+CREATE TABLE emp(dept_id INTEGER PRIMARY KEY, name VARCHAR, salary INTEGER);
+CREATE TABLE dept(id INTEGER PRIMARY KEY, name VARCHAR);
+
+SELECT emp.name, (SELECT dept.name FROM dept WHERE emp.dept_id=dept.id)
+FROM emp
+WHERE emp.salary > 1000;
+----
+We can see nested correlated subquery here, lets check the plan:
+[source,sql]
+----
+Project
+  CorrelatedNestedLoopJoin
+    Exchange
+      TableScan
+          table: PUBLIC.EMP
+          filters: >(SALARY, 1000)
+    ColocatedHashAggregate
+      Exchange
+        TableScan
+            table: PUBLIC.DEPT
+            filters: =($cor1.DEPT_ID, ID)
+----
+In Ignite 3 there is present some performance issues related to correlated subqueries and as a result:
+
+- Such subqueries may become bottlenecks.
+- Queries over even small tables can cause high CPU and memory consumption.
```

Comment:
In distributed system, such queries suffers from slow network calls, when Join node do remote index lookups. Imho, it is the major issue.